crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -0,0 +1,335 @@
+            ];
+            if let Some(p) = with_secondary_path {
+                typed_store::rocks::open_cf_opts_secondary(path, Some(&p), db_options, opt_cfs)
+            } else {
+                typed_store::rocks::open_cf_opts(path, db_options, opt_cfs)
+            }
+        }
+        .expect("Cannot open DB.");
+
+        let executed_sequence =
+            DBMap::reopen(&db, Some(EXEC_SEQ_TABLE_NAME)).expect("Cannot open CF.");
+
+        let (
+            objects,
+            owner_index,
+            transactions,
+            certificates,
+            pending_execution,
+            parent_sync,
+            effects,
+            assigned_object_versions,
+            next_object_versions,
+            consensus_message_processed,
+            batches,
+            last_consensus_index,
+            epochs,
+        ) = reopen! (
+            &db,
+            OBJECTS_TABLE_NAME;<ObjectKey, Object>,
+            OWNER_INDEX_TABLE_NAME;<(Owner, ObjectID), ObjectInfo>,
+            TX_TABLE_NAME;<TransactionDigest, TransactionEnvelope<S>>,
+            CERTS_TABLE_NAME;<TransactionDigest, CertifiedTransaction>,
+            PENDING_EXECUTION;<InternalSequenceNumber, TransactionDigest>,
+            PARENT_SYNC_TABLE_NAME;<ObjectRef, TransactionDigest>,
+            EFFECTS_TABLE_NAME;<TransactionDigest, TransactionEffectsEnvelope<S>>,
+            ASSIGNED_OBJECT_VERSIONS_TABLE_NAME;<(TransactionDigest, ObjectID), SequenceNumber>,
+            NEXT_OBJECT_VERSIONS_TABLE_NAME;<ObjectID, SequenceNumber>,
+            CONSENSUS_MESSAGE_PROCESSED_TABLE_NAME;<TransactionDigest, bool>,
+            BATCHES_TABLE_NAME;<TxSequenceNumber, SignedBatch>,
+            LAST_CONSENSUS_TABLE_NAME;<u64, ExecutionIndices>,
+            EPOCH_TABLE_NAME;<EpochId, AuthenticatedEpoch>
+        );
+
+        Self {
+            objects,
+            owner_index,
+            transactions,
+            certificates,
+            pending_execution,
+            parent_sync,
+            effects,
+            assigned_object_versions,
+            next_object_versions,
+            consensus_message_processed,
+            executed_sequence,
+            batches,
+            last_consensus_index,
+            epochs,
+        }
+    }
+
+    /// Open an authority store by directory path in read-write mode
+    pub fn open_read_write<P: AsRef<Path>>(path: P, db_options: Option<Options>) -> Self {
+        Self::open_impl(path, db_options, None)
+    }
+
+    /// Open an authority store by directory path in read only mode
+    pub fn open_read_only<P: AsRef<Path>>(
+        path: P,
+        secondary_path: P,
+        db_options: Option<Options>,
+    ) -> Self {
+        Self::open_impl(path, db_options, Some(secondary_path))
+    }
+
+    // TODO: condense with macros
+    pub fn dump(&self, table_name: &str) -> anyhow::Result<BTreeMap<String, String>> {
+        Ok(match table_name {
+            OBJECTS_TABLE_NAME => {
```

Comment:
Next steps: These will be auto-generated by a procedural macro for any table which used DBMap.